lib/router/mbaas-auth.js

```diff
@@ -0,0 +1,36 @@
+const q = require('q');
+
+/**
+ * Auth sends a request on 'wfm:user:auth' topic, raincatcher-demo-auth is subscriber
+ * of this topic and verifies passed in credentials.
+ *
+ * @param {Object} mediator object used to subscribe and publish read and auth topics
+ * @param {String} username Unique identifier of the user.
```

Comment:
The implementation on `demo-auth` has topics for an actual id (generated random string, but since we're going to use mongo as our primary storage they're going to map to ObjectIDs), and for usernames, it's just that this implementation is only using the `:username:read` topic since it's interested in validating the auth credentials, but I think the name is still valid.

We should definitely be wary of using the unique id for the user after its data is put in the session store, and a separate non-semantic unique id is probably important to keep for non-password authentication schemes.